src/compiler/program.ts

```diff
@@ -1582,11 +1583,11 @@ namespace ts {
         function emitWorker(program: Program, sourceFile: SourceFile | undefined, writeFileCallback: WriteFileCallback | undefined, cancellationToken: CancellationToken | undefined, emitOnlyDtsFiles?: boolean, customTransformers?: CustomTransformers, forceDtsEmit?: boolean): EmitResult {
             let declarationDiagnostics: readonly Diagnostic[] = [];
 
-            if (!forceDtsEmit) {
-                if (options.noEmit) {
-                    return { diagnostics: declarationDiagnostics, sourceMaps: undefined, emittedFiles: undefined, emitSkipped: true };
-                }
+            if (options.listFilesOnly || (!forceDtsEmit && shouldSuppressEmit(options))) {

```

Comment:
@sheetalkamat Is this okay?  This was the only place my new switch diverged from `--noEmit`.

Reply:
Moot.